service/history/timer_queue_standby_task_executor.go

```diff
@@ -124,7 +125,13 @@ func (t *timerQueueStandbyTaskExecutor) Execute(
 		err = t.executeDeleteHistoryEventTask(ctx, task)
 	case *tasks.StateMachineTimerTask:
 		err = t.executeStateMachineTimerTask(ctx, task, func(node *hsm.Node, task hsm.Task) error {
```

Comment:
You need to wrap this in `t.processTmer` so it can retry for a limited time same as you did for outbound tasks.

Reply:
I thought about it, but we can't exactly because `t.processTimer` does the stuff in the `t.executeStateMachineTimerTask`. I'll see what I can do to avoid duplication a lot of the code.